src/main/java/com/j256/ormlite/dao/BaseDaoImpl.java

```diff
@@ -57,7 +57,7 @@
  *            needs an ID parameter however so you can use Void or Object to satisfy the compiler.
  * @author graywatson
  */
-public abstract class BaseDaoImpl<T, ID> implements Dao<T, ID> {
+public class BaseDaoImpl<T, ID> implements Dao<T, ID> {
```

Comment:
Why was this change introduced? It doesn't appear that you require this change for your pr, so I suggest to undo.

Reply:
I'll undo it. But I was wondering why this was abstract in the first place.
Since it is abstract, one cannot simply instantiate it, but instead there are lots of anonymous classes:
```
new BaseDaoImpl<ForeignFoo, Integer>(connectionSource, ForeignFoo.class) {
				}
```

Why is that?

Reply:
Just because this pr doesn't require this change, and because it's untested and it's unclear what kind of repercussions it'll have in the wild. It's better to create a separate pr which introduces this change as part of a bigger refactor to remove abstract classes, for example.

Reply:
That's true. I've reverted this change.